atsign-foundation/at_c
# fix: Add atclient_pkam_authenticate_options and populate atserver host and…

… port when values are not set



**- What I did**
- Modify `atclient_pkam_authenticate` function to fetch the atserver host and port when not set by the user. 

**- How I did it**
- Introduce 'atclient_pkam_authenticate_options' structure which contains root server host and port and atserver host and port as data members.
  -  The root server host and port will be defaulted to the prod root server.
- Modify the signature of `atclient_pkam_authenticate` function to accept 'atclient_pkam_authenticate_options' and remove atserver_host and atserver_port.
- Refactor the examples to consume the `atclient_apkam_authenticate` function changes.

**- How to verify it**
- Tested manually and able to fetch the atserver host and port when not set by the user.
- After the changes, ran examples and works expected.
